lib/app/sequence_base.rb

```diff
@@ -253,6 +254,32 @@ def self.requires(*requires)
         @@requires[sequence_name] || []
       end
 
+      def self.new_requires(*requires)
```

Comment:
I vaguely remember some conversation on not adding `new_requires`, but instead updating `requires` to be able to handle when the requirement does not already exist.  the logic of this method could still be encapsulated in its own method that `requires` uses to make it easy the logic easier to follow.

I think that just updating `requires` would ultimately be a cleaner and less confusing solution for test writers.  Any big obstacles to that I'm not thinking of?

Reply:
> I think adding the labels and descriptions in the module.yml is a good way to start, but I'm worried about having to manage and update the prerequisites in multiple places. Right now we have a situation where they need to be edited in the ERBs, model and test code and its a bear to deal with.

Could take care of this using ruby.  By default have these just be defined completely in the sequences.  And if you use in multiple sequences place in a common ruby file / module and include the common file.  And/or you could have it be smart enough that only 1 sequence has to include the label/description portions, and in the other just require the parameter name it is good enough.

Reply:
We need to be optimizing for test writers.  I don't quite understand what 'new_requires' means by looking at it.  Is it stuff that is in only my module, and not defined in another module?

I personally think we should just consider things in 'requires' as 'global variables' that can be shared between modules.  If we want to be a little more explicit about where the variable is defined, we could add some kind of namespacing, so that if the SMART sequence defines "token" maybe to refer to it elsewhere you do `:requires SMART.token` or something.

Reply:
I think the namespacing could be cool.  Since we currently expect the requires to be accessible as methods on `@instance` though it might be a little tricky so I'd lean towards making that change in a future PR so we can at least have the baseline functionality in quick.

Reply:
Yeah, I should have explained in the PR, but `new_requires` is just a placeholder to show how the pattern would look without having to go back and change all the previous requires. Ultimately, it would just use `requires`